config/settings/production.py

```diff
@@ -1,301 +1,404 @@
+        "AUTOCOMMIT": False,
+    },
+    "robot": {
+        "USE_REDIS_CACHE": "default",
+        "DEFAULT_TIMEOUT": 7200,
+        "AUTOCOMMIT": False,
+    },
+}
 
-# Site URL: assumes appname.herokuapp.com
-SITE_URL = env("SITE_URL")
-FROM_EMAIL = env("FROM_EMAIL")
+# Site URL
+SITE_URL = None
+FROM_EMAIL = "test@mailinator.com"
 
 # Github credentials
-GITHUB_WEBHOOK_SECRET = env("GITHUB_WEBHOOK_SECRET")
+GITHUB_USERNAME = env("GITHUB_USERNAME", default=None)
+GITHUB_PASSWORD = env("GITHUB_PASSWORD", default=None)
+GITHUB_WEBHOOK_SECRET = None
 
 # Salesforce OAuth Connected App credentials
-CONNECTED_APP_CLIENT_ID = env("CONNECTED_APP_CLIENT_ID")
-CONNECTED_APP_CLIENT_SECRET = env("CONNECTED_APP_CLIENT_SECRET")
-CONNECTED_APP_CALLBACK_URL = env("CONNECTED_APP_CALLBACK_URL")
+CONNECTED_APP_CLIENT_ID = None
+CONNECTED_APP_CLIENT_SECRET = None
+CONNECTED_APP_CALLBACK_URL = None
+
+SFDX_CLIENT_ID = None
+SFDX_HUB_KEY = None
+SFDX_HUB_USERNAME = None
+
+# Application Behaviors
+GITHUB_STATUS_UPDATES_ENABLED = env.bool("GITHUB_STATUS_UPDATES_ENABLED", True)
+METACI_FLOW_CALLBACK_ENABLED = env.bool("METACI_FLOW_CALLBACK_ENABLED", True)
+
+# Number of scratch orgs to leave available in the org.
+SCRATCH_ORG_RESERVE = env.int("METACI_SCRATCH_ORG_RESERVE", 10)
+
+# Autoscaler class used for scaling the worker formation
+METACI_WORKER_AUTOSCALER = env(
+    "METACI_WORKER_AUTOSCALER", default="metaci.build.autoscaling.NonAutoscaler"
+)
+# What's the max number of worker dynos we should scale up to
+METACI_MAX_WORKERS = env.int("METACI_MAX_WORKERS", 3)
+# How many worker slots to reserve for high-priority jobs.
+# Should be less than METACI_MAX_WORKERS
+METACI_WORKER_RESERVE = env.int("METACI_WORKER_RESERVE", 1)
+WORKER_DYNO_NAME = env("WORKER_DYNO_NAME", default=None) or "worker"
+
+
+# Django REST Framework
+REST_FRAMEWORK = {
+    "DEFAULT_AUTHENTICATION_CLASSES": (
+        "rest_framework.authentication.BasicAuthentication",
+        "rest_framework.authentication.TokenAuthentication",
+        "rest_framework.authentication.SessionAuthentication",
+    ),
+    "DEFAULT_FILTER_BACKENDS": ("django_filters.rest_framework.DjangoFilterBackend",),
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.PageNumberPagination",
+    "PAGE_SIZE": 100,
+    "DEFAULT_PERMISSION_CLASSES": (
+        "metaci.api.permissions.IsOnSecureNetwork",
+        "rest_framework.permissions.IsAdminUser",
+    ),
+}
+
+# log_request_id settings
+LOG_REQUEST_ID_HEADER = "HTTP_X_REQUEST_ID"
+LOG_REQUESTS = True
+GENERATE_REQUEST_ID_IF_NOT_IN_HEADER = True
+REQUEST_ID_RESPONSE_HEADER = "X-Request-ID"
 
-SFDX_CLIENT_ID = env("SFDX_CLIENT_ID")
-SFDX_HUB_KEY = env("SFDX_HUB_KEY")
-SFDX_HUB_USERNAME = env("SFDX_HUB_USERNAME")
+# django-guardian settings
+GUARDIAN_MONKEY_PATCH = False
 
-# django-defender configuration
-DEFENDER_REDIS_NAME = "default"
+JS_REVERSE_JS_VAR_NAME = "api_urls"
+JS_REVERSE_EXCLUDE_NAMESPACES = ["admin", "admin_rest"]
```

Comment:
@jofsky why are there so many changes in production.py? I would expect to see minimal changes here since we are not switching production to use docker at this time.

Reply:
So i had to use --no-verify on the commit to get it to pass with the current production.py file

Reply:
That's normal -- since it uses "import *", it's not compliant with our flake8 checks. So we usually just manually review to make sure there aren't _other_ flake8 errors, and then check it in with --no-verify.

There's probably a better way, but replacing all of production.py is DEFINITELY not a good solution. Extending from a common set of base settings is a pattern we use in all our projects that has worked well for us. I'm willing to discuss ways we might change that, but let's consider it separately from the docker changes for robot.